File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */


int getDecimalValue(struct ListNode* head){
    struct ListNode* temp=head;
    int count = 0;
    int total =0;
    while(temp!=NULL){
        count++;
        temp = temp->next;
    }
    temp = head;
    int i = count -1;
    while(temp!=NULL){
        total +=(pow(2,i)*(temp->val));
        i--;
        temp = temp->next;
    }
    return total;
}
    
    

File: 1290-convert-binary-number-in-a-linked-list-to-integer/1290-convert-binary-number-in-a-linked-list-to-integer.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def getDecimalValue(self, head: ListNode) -> int:
        total = 0
        while(head):
            total = total * 2 + head.val
            head = head.next
        return total